Index: asterisk-1.2.24.dfsg+pf.0.1~svn1329/term.c
===================================================================
--- asterisk-1.2.24.dfsg+pf.0.1~svn1329.orig/term.c	2007-08-08 14:42:37.000000000 +0200
+++ asterisk-1.2.24.dfsg+pf.0.1~svn1329/term.c	2007-08-08 14:43:22.000000000 +0200
@@ -79,8 +79,8 @@
 	int termfd = -1, parseokay = 0, i;
 
 	if (!term)
-		return 0;
-	if (!option_console || option_nocolor || !option_nofork)
+		term="xterm";
+	if (option_nocolor)
 		return 0;
 
 	for (i=0 ;; i++) {
