Index: asterisk-1.2.24.dfsg+pf.0.1~svn1329/channels/h323/Makefile
===================================================================
--- asterisk-1.2.24.dfsg+pf.0.1~svn1329.orig/channels/h323/Makefile	2007-08-08 14:42:37.000000000 +0200
+++ asterisk-1.2.24.dfsg+pf.0.1~svn1329/channels/h323/Makefile	2007-08-08 14:43:25.000000000 +0200
@@ -21,8 +21,7 @@
 OPENH323DIR=$(HOME)/openh323
 endif
 
-clean:
-	rm -f *.so *.o .depend
+-include $(OPENH323DIR)/openh323u.mak
 
 $(SOURCES)::	$(SOURCES:.cxx=.cpp)
 	ln -f $< $@
@@ -31,7 +30,6 @@
 	touch $(SOURCES)
 
 libchanh323.a:	$(OBJS)
-	include $(OPENH323DIR)/openh323u.mak
 	ar crv $@ $(OBJS)
 
 Makefile.ast:	FORCE
