Index: asterisk-1.2.24.dfsg+pf.0.1~svn1329/Makefile
===================================================================
--- asterisk-1.2.24.dfsg+pf.0.1~svn1329.orig/Makefile	2007-08-08 14:43:02.000000000 +0200
+++ asterisk-1.2.24.dfsg+pf.0.1~svn1329/Makefile	2007-08-08 14:43:17.000000000 +0200
@@ -19,8 +19,8 @@
 # CROSS_COMPILE=/opt/montavista/pro/devkit/arm/xscale_be/bin/xscale_be-
 # CROSS_COMPILE_BIN=/opt/montavista/pro/devkit/arm/xscale_be/bin/
 # CROSS_COMPILE_TARGET=/opt/montavista/pro/devkit/arm/xscale_be/target
-CC=$(CROSS_COMPILE)gcc
-HOST_CC=gcc
+#CC=$(CROSS_COMPILE)gcc
+HOST_CC=$(CC)
 # CROSS_ARCH=Linux
 # CROSS_PROC=arm
 # SUB_PROC=xscale # or maverick
