Index: asterisk-1.2.23~dfsg/Makefile
===================================================================
--- asterisk-1.2.23~dfsg.orig/Makefile	2007-08-06 12:02:19.000000000 +0200
+++ asterisk-1.2.23~dfsg/Makefile	2007-08-06 12:02:20.000000000 +0200
@@ -537,7 +537,7 @@
 	for x in $(SUBDIRS); do $(MAKE) -C $$x || exit 1 ; done
 
 clean:
-	for x in $(SUBDIRS); do $(MAKE) -C $$x clean || exit 1 ; done
+	for x in $(SUBDIRS); do $(MAKE) -C $$x clean || true ; done
 	rm -f *.o *.so asterisk .depend
 	rm -f defaults.h
 	rm -f include/asterisk/build.h
