Index: asterisk-1.2.24.dfsg+pf.0.1~svn1329/asterisk.c
===================================================================
--- asterisk-1.2.24.dfsg+pf.0.1~svn1329.orig/asterisk.c	2007-06-28 01:22:13.000000000 +0200
+++ asterisk-1.2.24.dfsg+pf.0.1~svn1329/asterisk.c	2007-08-08 14:30:20.000000000 +0200
@@ -155,6 +155,7 @@
 int option_nofork=0;			/*!< Do not fork */
 int option_quiet=0;			/*!< Keep quiet */
 int option_console=0;			/*!< Console mode, no background */
+int option_daemonize=0;		/*!< Daemonize. Ever if -v or -d */
 int option_highpriority=0;		/*!< Run in realtime Linux priority */
 int option_remote=0;			/*!< Remote CLI */
 int option_exec=0;			/*!< */
@@ -1849,6 +1850,7 @@
 	printf("   -G <group>      Run as a group other than the caller\n");
 	printf("   -U <user>       Run as a user other than the caller\n");
 	printf("   -c              Provide console CLI\n");
+	printf("   -F              Force Fork even if -v or -d were given\n");
 	printf("   -d              Enable extra debugging\n");
 	printf("   -f              Do not fork\n");
 	printf("   -g              Dump core in case of a crash\n");
@@ -2085,7 +2087,7 @@
 	}
 	*/
 	/* Check for options */
-	while((c=getopt(argc, argv, "tThfdvVqprRgcinx:U:G:C:L:M:")) != -1) {
+	while((c=getopt(argc, argv, "tThfdvVqprRgcFinx:U:G:C:L:M:")) != -1) {
 		switch(c) {
 		case 'd':
 			option_debug++;
@@ -2095,6 +2097,9 @@
 			option_console++;
 			option_nofork++;
 			break;
+		case 'F':
+			option_daemonize++;
+			break;
 		case 'f':
 			option_nofork++;
 			break;
@@ -2307,7 +2312,9 @@
 	} else
 		ast_log(LOG_WARNING, "Unable to open pid file '%s': %s\n", (char *)ast_config_AST_PID, strerror(errno));
 
-	if (!option_verbose && !option_debug && !option_nofork && !option_console) {
+	if (option_daemonize ||
+			(!option_verbose && !option_debug && !option_nofork && !option_console)
+	) {
 		daemon(0,0);
 		/* Blindly re-write pid file since we are forking */
 		unlink((char *)ast_config_AST_PID);
