Index: asterisk-1.4.26/main/channel.c
===================================================================
--- asterisk-1.4.26.orig/main/channel.c	2009-07-22 18:21:23.000000000 +0200
+++ asterisk-1.4.26/main/channel.c	2009-07-22 18:55:12.000000000 +0200
@@ -2576,6 +2576,7 @@
 	case AST_CONTROL_ANSWER:
 	case AST_CONTROL_HANGUP:
 	case AST_CONTROL_CONNECTEDLINE:
+	case AST_CONTROL_ATXFERCMD:
 		return 0;
 
 	case AST_CONTROL_CONGESTION:
@@ -2678,6 +2679,7 @@
 	case AST_CONTROL_RING:
 	case AST_CONTROL_HOLD:
 	case AST_CONTROL_UNHOLD:
+	case AST_CONTROL_ATXFERCMD:
 		/* Nothing left to do for these. */
 		res = 0;
 		break;
