Index: asterisk-1.4.26.2/channels/chan_misdn.c
===================================================================
--- asterisk-1.4.26.2.orig/channels/chan_misdn.c	2009-10-21 20:05:17.000000000 +0200
+++ asterisk-1.4.26.2/channels/chan_misdn.c	2009-10-21 20:06:20.000000000 +0200
@@ -4766,6 +4766,7 @@
 		if (!misdn_cap_is_speech(ch->bc->capability)) {
 			struct ast_frame frame;
 			/*In Data Modes we queue frames*/
+			memset(&frame, 0, sizeof(frame));
 			frame.frametype  = AST_FRAME_VOICE; /*we have no data frames yet*/
 			frame.subclass = AST_FORMAT_ALAW;
 			frame.datalen = bc->bframe_len;
