Index: asterisk-1.4.41/main/channel.c
===================================================================
--- asterisk-1.4.41.orig/main/channel.c	2011-05-06 17:08:54.000000000 +0200
+++ asterisk-1.4.41/main/channel.c	2011-05-06 17:08:55.000000000 +0200
@@ -4116,6 +4116,13 @@
 	struct ast_var_t *current, *newvar;
 	const char *varname;
 
+        if ((parent != NULL) && (child != NULL))
+                manager_event(EVENT_FLAG_CALL, "Inherit",
+                              "Parent: %s\r\n"
+                              "Child: %s\r\n",
+                              parent->name,
+                              child->name);
+
 	AST_LIST_TRAVERSE(&parent->varshead, current, entries) {
 		int vartype = 0;
 
