diff -Nurp asterisk-1.2.10.orig/build_tools/mkdep asterisk-1.2.10/build_tools/mkdep
--- asterisk-1.2.10.orig/build_tools/mkdep	2005-06-20 19:26:08.000000000 +0200
+++ asterisk-1.2.10/build_tools/mkdep	2006-07-28 12:38:03.000000000 +0200
@@ -90,11 +90,11 @@ umask $um
 trap 'rm -rf $DTMP ; trap 2 ; kill -2 $$' 1 2 3 13 15
 
 if [ x$pflag = xp ]; then
-	${CC:-cc} -M "$@" 2>/dev/null | sed -e 's;\.o :; :;' -e 's; \./; ;g' > $TMP
+	${CC:-cc} -MM "$@" 2>/dev/null | sed -e 's;\.o :; :;' -e 's; \./; ;g' > $TMP
 elif [ x$dflag = xd ]; then
-	${CC:-cc} -M "$@" 2>/dev/null | sed -e 's;\.o:;\.o::;' -e 's; \./; ;g' > $TMP
+	${CC:-cc} -MM "$@" 2>/dev/null | sed -e 's;\.o:;\.o::;' -e 's; \./; ;g' > $TMP
 else
-	${CC:-cc} -M "$@" 2>/dev/null | sed -e 's; \./; ;g' > $TMP
+	${CC:-cc} -MM "$@" 2>/dev/null | sed -e 's; \./; ;g' > $TMP
 fi
 
 if [ $? != 0 ]; then
