diff -Nurp busybox-1.1.3.orig/networking/httpd.c busybox-1.1.3/networking/httpd.c
--- busybox-1.1.3.orig/networking/httpd.c	2006-03-22 22:16:19.000000000 +0100
+++ busybox-1.1.3/networking/httpd.c	2006-07-10 18:47:24.000000000 +0200
@@ -1815,9 +1815,27 @@ FORBIDDEN:      /* protect listing /cgi-
 			goto FORBIDDEN;     // protect listing cgi-bin/
 		sendCgi(url, prequest, length, cookie, content_type);
     } else {
-	if (prequest != request_GET)
+#ifdef CONFIG_FEATURE_HTTPD_CONFIG_WITH_SCRIPT_INTERPR
+	char *suffix = strrchr(url, '.');
+	int interpr_cgi = 0;
+
+	if(suffix) {
+	    Htaccess * cur;
+	    size_t suffix_len = strlen(suffix);
+	    for (cur = config->script_i; cur; cur = cur->next)
+		if(strncmp(cur->before_colon + 1, suffix, suffix_len) == 0) {
+		    interpr_cgi = 1;
+		    break;
+		}
+	}
+
+	if (interpr_cgi) {
+	    sendCgi(url, prequest, length, cookie, content_type);
+	} else {
+#endif  /* CONFIG_FEATURE_HTTPD_CONFIG_WITH_SCRIPT_INTERPR */
+	    if (prequest != request_GET)
 		sendHeaders(HTTP_NOT_IMPLEMENTED);
-	else {
+	    else {
 #endif  /* CONFIG_FEATURE_HTTPD_CGI */
 		if(purl[-1] == '/')
 			strcpy(purl, "index.html");
@@ -1832,9 +1850,12 @@ FORBIDDEN:      /* protect listing /cgi-
 #endif
 
 #ifdef CONFIG_FEATURE_HTTPD_CGI
+	    }
+#ifdef CONFIG_FEATURE_HTTPD_CONFIG_WITH_SCRIPT_INTERPR
 	}
+#endif /* CONFIG_FEATURE_HTTPD_CONFIG_WITH_SCRIPT_INTERPR */
     }
-#endif
+#endif /* CONFIG_FEATURE_HTTPD_CGI */
 
   } while (0);
 
