diff -Nurp asterisk-addons-1.2.4.orig/asterisk-ooh323c/config/config.sub asterisk-addons-1.2.4/asterisk-ooh323c/config/config.sub
--- asterisk-addons-1.2.4.orig/asterisk-ooh323c/config/config.sub	2006-05-31 18:50:32.000000000 +0200
+++ asterisk-addons-1.2.4/asterisk-ooh323c/config/config.sub	2006-09-14 14:48:21.000000000 +0200
@@ -118,7 +118,7 @@ esac
 # Here we must recognize all the valid KERNEL-OS combinations.
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
-  nto-qnx* | linux-gnu* | freebsd*-gnu* | netbsd*-gnu* | storm-chaos* | os2-emx* | rtmk-nova*)
+  nto-qnx* | linux-gnu* | linux-uclibc* | freebsd*-gnu* | netbsd*-gnu* | storm-chaos* | os2-emx* | rtmk-nova*)
     os=-$maybe_os
     basic_machine=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`
     ;;
@@ -1108,7 +1108,7 @@ case $os in
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -linux-gnu* | -uxpv* | -beos* | -mpeix* | -udk* \
+	      | -mingw32* | -linux-gnu* | -linux-uclibc* | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
diff -Nurp asterisk-addons-1.2.4.orig/asterisk-ooh323c/configure asterisk-addons-1.2.4/asterisk-ooh323c/configure
--- asterisk-addons-1.2.4.orig/asterisk-ooh323c/configure	2006-05-31 18:50:32.000000000 +0200
+++ asterisk-addons-1.2.4/asterisk-ooh323c/configure	2006-09-14 15:07:40.000000000 +0200
@@ -3348,7 +3348,7 @@ irix5* | irix6* | nonstopux*)
   ;;
 
 # This must be Linux ELF.
-linux-gnu*)
+linux-gnu*|linux-uclibc*)
   case $host_cpu in
   alpha* | hppa* | i*86 | mips | mipsel | powerpc* | sparc* | ia64* | s390* | x86_64*)
     lt_cv_deplibs_check_method=pass_all ;;
@@ -4720,6 +4720,7 @@ fi
 # Transform linux* to *-*-linux-gnu*, to support old configure scripts.
 case $host_os in
 linux-gnu*) ;;
+linux-uclibc*) ;;
 linux*) host=`echo $host | sed 's/^\(.*-.*-linux\)\(.*\)$/\1-gnu\2/'`
 esac
 
@@ -6261,7 +6262,7 @@ linux-gnuoldld* | linux-gnuaout* | linux
   ;;
 
 # This must be Linux ELF.
-linux-gnu*)
+linux-gnu*|linux-uclibc*)
   version_type=linux
   need_lib_prefix=no
   need_version=no
