diff -Nurp chan_misdn.orig/Makefile chan_misdn/Makefile
--- chan_misdn.orig/Makefile	2006-08-08 14:15:26.000000000 +0200
+++ chan_misdn/Makefile	2006-09-04 17:03:23.000000000 +0200
@@ -14,7 +14,8 @@ CANDIDATE_VERSION=0.$(MAJOR).$(MINOR_CAN
 # The following line tells the makefile where to install the module, 
 # if necessary, please edit it 
 #
-INSTALL_MODPATH=/usr/lib/asterisk/modules
+INSTALL_PREFIX?=
+INSTALL_MODPATH=$(INSTALL_PREFIX)/usr/lib/asterisk/modules
 
 
 AST_PACK_DIR := /usr/src/ast-ci-svn/0.4.0
@@ -30,7 +31,9 @@ MEMPATROL= #-lelf -lmpatrol
 
 MISDNOBJS=chan_misdn.o misdn/chan_misdn_lib.a misdn_config.o 
 
-ADDOBJS+=/usr/lib/libisdnnet.so /usr/lib/libmISDN.so
+ADDOBJS+=$(INSTALL_PREFIX)/usr/lib/libisdnnet.so
+ADDOBJS+=$(INSTALL_PREFIX)/usr/lib/libmISDN.so
+
 
 all: chan_misdn.so 
 
@@ -38,7 +41,7 @@ $(ADDOBJS):
 	@if [ ! -f $(MISDNUSERLIB)/libisdnnet.so -o ! -f $(MISDNUSERLIB)/libmISDN.so ] ; then echo -e "\nYou need mISDNuser to compile chan_misdn!\nPlease read the README\n" ; exit 1;  fi 
 
 chan_misdn.so: $(MISDNOBJS) $(ADDOBJS) 
-	$(CC) -shared -Xlinker -x -L/usr/lib/mISDNuser -o $@ $(MISDNOBJS) -lisdnnet -lmISDN $(MEMPATROL)
+	$(CC) -fPIC -shared -Xlinker -x -L$(INSTALL_PREFIX)/usr/lib/mISDNuser -o $@ $(MISDNOBJS) -lisdnnet -lmISDN $(MEMPATROL)
 
 chan_misdn.o: chan_misdn.c
 	$(CC) $(CFLAGS) -DCHAN_MISDN_VERSION=\"$(CANDIDATE_VERSION)\" -c chan_misdn.c
@@ -52,8 +55,8 @@ misdn/chan_misdn_lib.a: misdn/isdn_lib.c
 
 install: chan_misdn.so 
 	install -m 644 chan_misdn.so $(INSTALL_MODPATH)/
-	if [ ! -f /etc/asterisk/misdn.conf ] ; then cp misdn.conf /etc/asterisk/; fi
-	if [ ! -e /dev/mISDN ] ; then mknod /dev/mISDN c 46 0; fi
+	#if [ ! -f /etc/asterisk/misdn.conf ] ; then cp misdn.conf /etc/asterisk/; fi
+	#if [ ! -e /dev/mISDN ] ; then mknod /dev/mISDN c 46 0; fi
 	@echo 
 	@echo INSTALLATION SUCCEEDED
 
