diff -Nurp mISDN-CVS-2006-08-26.orig/Makefile mISDN-CVS-2006-08-26/Makefile
--- mISDN-CVS-2006-08-26.orig/Makefile	2006-08-25 13:50:41.000000000 +0200
+++ mISDN-CVS-2006-08-26/Makefile	2006-09-04 14:43:05.000000000 +0200
@@ -1,17 +1,19 @@
 BASEDIR=$(shell pwd)
 
 
-INSTALL_PREFIX := /
+INSTALL_PREFIX ?= /
 export INSTALL_PREFIX
 
 #PATH to linux source/headers
 #LINUX=/usr/src/linux
 MODS=/lib/modules/$(shell uname -r)
-LINUX=$(MODS)/build
-LINUX_SOURCE=$(MODS)/source
-UPDATE_MODULES=$(shell which update-modules)
-MODULES_UPDATE=$(shell which modules-update)
-DEPMOD=$(shell which depmod)
+LINUX?=$(MODS)/build
+LINUX_SOURCE?=$(MODS)/source
+export CROSS_COMPILE
+
+#UPDATE_MODULES=$(shell which update-modules)
+#MODULES_UPDATE=$(shell which modules-update)
+#DEPMOD=$(shell which depmod)
 
 
 MISDNDIR=$(BASEDIR)
@@ -25,7 +27,7 @@ CONFIGS+=CONFIG_MISDN_DRV=m 
 CONFIGS+=CONFIG_MISDN_DSP=m 
 CONFIGS+=CONFIG_MISDN_HFCMULTI=m 
 CONFIGS+=CONFIG_MISDN_HFCPCI=m
-CONFIGS+=CONFIG_MISDN_HFCUSB=m
+#CONFIGS+=CONFIG_MISDN_HFCUSB=m
 CONFIGS+=CONFIG_MISDN_XHFC=m
 CONFIGS+=CONFIG_MISDN_HFCMINI=m
 CONFIGS+=CONFIG_MISDN_W6692=m
@@ -54,18 +56,18 @@ install: all
 	mkdir -p $(INSTALL_PREFIX)/usr/include/linux/
 	cp $(MISDNDIR)/include/linux/*.h $(INSTALL_PREFIX)/usr/include/linux/
 	mkdir -p $(INSTALL_PREFIX)/usr/sbin/
-	install -m755 misdn-init $(INSTALL_PREFIX)/usr/sbin/
-	if [ -d $(INSTALL_PREFIX)/etc/init.d ]; then \
-		if [ -e $(INSTALL_PREFIX)/etc/init.d/misdn-init ]; then rm -rf $(INSTALL_PREFIX)/etc/init.d/misdn-init; fi; \
-		ln -s $(INSTALL_PREFIX)/usr/sbin/misdn-init $(INSTALL_PREFIX)/etc/init.d/misdn-init; \
-	fi
+	#install -m755 misdn-init $(INSTALL_PREFIX)/usr/sbin/
+	#if [ -d $(INSTALL_PREFIX)/etc/init.d ]; then \
+	#	if [ -e $(INSTALL_PREFIX)/etc/init.d/misdn-init ]; then rm -rf $(INSTALL_PREFIX)/etc/init.d/misdn-init; fi; \
+	#	ln -s $(INSTALL_PREFIX)/usr/sbin/misdn-init $(INSTALL_PREFIX)/etc/init.d/misdn-init; \
+	#fi
 	mkdir -p $(INSTALL_PREFIX)/etc/modprobe.d
 	cp mISDN.modprobe.d $(INSTALL_PREFIX)/etc/modprobe.d/mISDN
 	mkdir -p $(INSTALL_PREFIX)/etc/modules.d
 	cp mISDN.modprobe.d $(INSTALL_PREFIX)/etc/modules.d/mISDN
-	$(DEPMOD) 
-	$(UPDATE_MODULES)
-	$(MODULES_UPDATE)
+	#$(DEPMOD) 
+	#$(UPDATE_MODULES)
+	#$(MODULES_UPDATE)
 
 test_old_misdn:
 	@if echo -ne "#include <linux/mISDNif.h>" | gcc -C -E - 2>/dev/null 1>/dev/null  ; then \
