diff -Nurp mtd-utils-1.0.0.orig/Makefile mtd-utils-1.0.0/Makefile
--- mtd-utils-1.0.0.orig/Makefile	2006-04-30 23:27:57.000000000 +0200
+++ mtd-utils-1.0.0/Makefile	2006-09-11 17:16:30.000000000 +0200
@@ -10,13 +10,16 @@ INCLUDEDIR=/usr/include
 CC := $(CROSS)gcc
 CFLAGS := -I./include -O2 -Wall
 
-TARGETS = ftl_format flash_erase flash_eraseall nanddump doc_loadbios \
-	mkfs.jffs ftl_check mkfs.jffs2 flash_lock flash_unlock flash_info \
-	flash_otp_info flash_otp_dump mtd_debug flashcp nandwrite \
-	jffs2dump \
-	nftldump nftl_format docfdisk \
-	rfddump rfdformat \
-	sumtool #jffs2reader
+#TARGETS = ftl_format flash_erase flash_eraseall nanddump doc_loadbios \
+#	mkfs.jffs ftl_check mkfs.jffs2 flash_lock flash_unlock flash_info \
+#	flash_otp_info flash_otp_dump mtd_debug flashcp nandwrite \
+#	jffs2dump \
+#	nftldump nftl_format docfdisk \
+#	rfddump rfdformat \
+#	sumtool #jffs2reader
+
+TARGETS = flash_erase flash_eraseall flash_lock flash_unlock flash_info \
+	flash_otp_info flash_otp_dump flashcp
 
 SYMLINKS =
 
