diff -Nurp openh323_v1_18_0.orig/config.sub openh323_v1_18_0/config.sub
--- openh323_v1_18_0.orig/config.sub	2005-03-19 04:38:27.000000000 +0100
+++ openh323_v1_18_0/config.sub	2006-09-15 11:41:10.000000000 +0200
@@ -118,7 +118,7 @@ esac
 # Here we must recognize all the valid KERNEL-OS combinations.
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
-  nto-qnx* | linux-gnu* | linux-dietlibc | kfreebsd*-gnu* | netbsd*-gnu* | storm-chaos* | os2-emx* | rtmk-nova*)
+  nto-qnx* | linux-gnu* | linux-uclibc* | linux-dietlibc | kfreebsd*-gnu* | netbsd*-gnu* | storm-chaos* | os2-emx* | rtmk-nova*)
     os=-$maybe_os
     basic_machine=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`
     ;;
@@ -1137,7 +1137,7 @@ case $os in
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -linux-gnu* | -uxpv* | -beos* | -mpeix* | -udk* \
+	      | -mingw32* | -linux-gnu* | -linux-uclibc* | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
diff -Nurp openh323_v1_18_0.orig/configure openh323_v1_18_0/configure
--- openh323_v1_18_0.orig/configure	2005-08-30 03:12:38.000000000 +0200
+++ openh323_v1_18_0/configure	2006-09-15 11:41:10.000000000 +0200
@@ -3080,10 +3080,10 @@ echo $ECHO_N "checking for ${PWLIBDIR}/v
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "${PWLIBDIR}/version.h"; then
   eval "$as_ac_File=yes"
 else
@@ -3104,10 +3104,10 @@ echo $ECHO_N "checking for ${OPENH323DIR
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "${OPENH323DIR}/../pwlib/version.h"; then
   eval "$as_ac_File=yes"
 else
@@ -3129,10 +3129,10 @@ echo $ECHO_N "checking for ${HOME}/pwlib
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "${HOME}/pwlib/include/ptlib.h"; then
   eval "$as_ac_File=yes"
 else
@@ -3153,10 +3153,10 @@ echo $ECHO_N "checking for /usr/local/in
 if test "${ac_cv_file__usr_local_include_ptlib_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "/usr/local/include/ptlib.h"; then
   ac_cv_file__usr_local_include_ptlib_h=yes
 else
@@ -3215,10 +3215,10 @@ echo $ECHO_N "checking for /usr/include/
 if test "${ac_cv_file__usr_include_ptlib_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "/usr/include/ptlib.h"; then
   ac_cv_file__usr_include_ptlib_h=yes
 else
@@ -3985,10 +3985,10 @@ echo $ECHO_N "checking for /usr/local/in
 if test "${ac_cv_file__usr_local_include_ffmpeg_avcodec_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "/usr/local/include/ffmpeg/avcodec.h"; then
   ac_cv_file__usr_local_include_ffmpeg_avcodec_h=yes
 else
@@ -4193,10 +4193,10 @@ echo $ECHO_N "checking for /usr/local/li
 if test "${ac_cv_file__usr_local_lib_libvich263_so+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "/usr/local/lib/libvich263.so"; then
   ac_cv_file__usr_local_lib_libvich263_so=yes
 else
@@ -4219,10 +4219,10 @@ echo $ECHO_N "checking for /usr/local/in
 if test "${ac_cv_file__usr_local_include_vich263_encoder_h263_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
 if test -r "/usr/local/include/vich263/encoder-h263.h"; then
   ac_cv_file__usr_local_include_vich263_encoder_h263_h=yes
 else
diff -Nurp openh323_v1_18_0.orig/Makefile.in openh323_v1_18_0/Makefile.in
--- openh323_v1_18_0.orig/Makefile.in	2005-08-04 21:31:59.000000000 +0200
+++ openh323_v1_18_0/Makefile.in	2006-09-15 11:41:10.000000000 +0200
@@ -351,8 +351,8 @@ install:
 		ln -sf $(OH323_FILE).@MAJOR_VERSION@.@MINOR_VERSION@.@BUILD_NUMBER@ $(OH323_FILE) \
 	)
 	(cd $(DESTDIR)$(LIBDIR); \
-		rm -f  libopenh323.@SHAREDLIBEXT@ ; \
-		ln -sf $(OH323_FILE) libopenh323.@SHAREDLIBEXT@ \
+		rm -f  libopenh323.so ; \
+		ln -sf $(OH323_FILE) libopenh323.so \
 	)
 	( for dir in $(DESTDIR)$(LIBDIR) \
 		     $(DESTDIR)$(PREFIX)/include/openh323 \
diff -Nurp openh323_v1_18_0.orig/src/h323ep.cxx openh323_v1_18_0/src/h323ep.cxx
--- openh323_v1_18_0.orig/src/h323ep.cxx	2005-11-25 04:42:04.000000000 +0100
+++ openh323_v1_18_0/src/h323ep.cxx	2006-09-15 11:41:10.000000000 +0200
@@ -1038,7 +1038,7 @@ H323EndPoint::H323EndPoint()
 
 #else
 
-#  ifdef H323_AUDIO_CODECS
+#  if defined(H323_AUDIO_CODECS) && defined(P_AUDIO)
      // Should only need double buffering for Unix platforms
      soundChannelBuffers = 2;
 #  endif
