diff -Nurp res_sqlite.orig/Makefile res_sqlite/Makefile
--- res_sqlite.orig/Makefile	2006-08-23 14:43:21.000000000 +0200
+++ res_sqlite/Makefile	2006-09-11 18:30:26.000000000 +0200
@@ -3,7 +3,7 @@
 # export CC=gcc
 
 # you can override default options by calling make OPTIONS=...
-OPTIONS ?= -O2
+OPTIONS ?= -O2 -fPIC
 CFLAGS = -D _GNU_SOURCE -Wall -Wmissing-prototypes -Wstrict-prototypes $(OPTIONS)
 LIBS = -lpthread -lsqlite
 
