diff -Nurp vsftpd-2.0.5.orig/builddefs.h vsftpd-2.0.5/builddefs.h
--- vsftpd-2.0.5.orig/builddefs.h	2006-07-02 23:53:47.000000000 +0200
+++ vsftpd-2.0.5/builddefs.h	2006-07-11 13:57:14.000000000 +0200
@@ -2,7 +2,7 @@
 #define VSF_BUILDDEFS_H
 
 #undef VSF_BUILD_TCPWRAPPERS
-#define VSF_BUILD_PAM
+#undef VSF_BUILD_PAM
 #undef VSF_BUILD_SSL
 
 #endif /* VSF_BUILDDEFS_H */
diff -Nurp vsftpd-2.0.5.orig/Makefile vsftpd-2.0.5/Makefile
--- vsftpd-2.0.5.orig/Makefile	2006-07-03 14:25:41.000000000 +0200
+++ vsftpd-2.0.5/Makefile	2006-07-11 14:00:28.000000000 +0200
@@ -5,7 +5,7 @@ IFLAGS  = -idirafter dummyinc
 #CFLAGS = -g
 CFLAGS	=	-O2 -Wall -W -Wshadow #-pedantic -Werror -Wconversion
 
-LIBS	=	`./vsf_findlibs.sh`
+LIBS	=	-lcrypt
 LINK	=	-Wl,-s
 
 OBJS	=	main.o utility.o prelogin.o ftpcmdio.o postlogin.o privsock.o \
