diff -Nurp zaptel-1.2.10.orig/Makefile zaptel-1.2.10/Makefile
--- zaptel-1.2.10.orig/Makefile	2006-10-11 19:22:10.000000000 +0200
+++ zaptel-1.2.10/Makefile	2006-11-06 17:39:26.000000000 +0100
@@ -6,7 +6,14 @@
 #
 
 
-HOSTCC=gcc
+INSTALL_PREFIX:=$(DESTDIR)$(PREFIX)
+BUILDCC=cc
+BUILDAR=ar
+HOSTCC?=$(BUILDCC)
+HOSTAR?=$(BUILDAR)
+ARCH?=$(shell uname -m)
+CC:=$(HOSTCC)
+
 PWD:=$(shell pwd)
 
 # If you want to build for a kernel other than the current kernel, set KVERS
@@ -24,8 +31,8 @@ endif
 KINCLUDES:=$(KSRC)/include
 
 CFLAGS+=-I. -O4 -g -Wall -DBUILDING_TONEZONE #-DTONEZONE_DRIVER
-CFLAGS_PPC:=$(shell if uname -m | grep -q ppc; then echo "-fsigned-char"; fi)
-CFLAGS_X86-64:=$(shell if uname -m | grep -q x86_64; then echo "-m64"; fi)
+CFLAGS_PPC:=$(shell if echo $(ARCH) | grep -q ppc; then echo "-fsigned-char"; fi)
+CFLAGS_X86-64:=$(shell if echo $(ARCH) | grep -q x86_64; then echo "-m64"; fi)
 CFLAGS+=$(CFLAGS_PPC) $(CFLAGS_X86-64)
 LCFLAGS:=-fPIC $(CFLAGS) -DBUILDING_TONEZONE
 KFLAGS:=-I$(KINCLUDES) -O6
@@ -34,9 +41,9 @@ KFLAGS+=-DMODULE -D__KERNEL__ -DEXPORT_S
 ifneq (,$(wildcard $(KINCLUDES)/linux/modversions.h))
   KFLAGS+=-DMODVERSIONS -include $(KINCLUDES)/linux/modversions.h
 endif
-KFLAGS_PPC:=$(shell if uname -m | grep -q ppc; then echo "-msoft-float -fsigned-char"; fi)
+KFLAGS_PPC:=$(shell if echo $(ARCH) | grep -q ppc; then echo "-msoft-float -fsigned-char"; fi)
 KFLAGS+=$(KFLAGS_PPC)
-KFLAGS+=$(shell if uname -r | grep -q 2.4; then if uname -m | grep -q x86_64; then echo "-mcmodel=kernel"; fi; fi)
+KFLAGS+=$(shell if echo $(KVERS) | grep -q 2.4; then if uname -m | grep -q x86_64; then echo "-mcmodel=kernel"; fi; fi)
 
 #
 # Features are now configured in zconfig.h
@@ -47,10 +54,10 @@ CFLAGS+=-DSTANDALONE_ZAPATA
 KMAKE:= $(MAKE) -C $(KSRC) SUBDIRS=$(PWD)
 KMAKE_INST:= $(KMAKE) INSTALL_MOD_PATH=$(INSTALL_PREFIX) INSTALL_MOD_DIR=misc modules_install
 
-ROOT_PREFIX=
-INSTALL_PREFIX:=$(DESTDIR)
+#ROOT_PREFIX=
+#INSTALL_PREFIX:=$(DESTDIR)
 
-CONFIG_FILE:=$(INSTALL_PREFIX)/etc/zaptel.conf
+CONFIG_FILE:=$(PREFIX)/etc/zaptel.conf
 CFLAGS+=-DZAPTEL_CONFIG=\"$(CONFIG_FILE)\"
 
 ifeq (2.6,$(shell echo $(KVERS) | cut -d. -f1-2))
@@ -65,7 +72,7 @@ DYNFS:=$(shell ps ax | grep -v grep | gr
 endif
 
 ifeq ($(BUILDVER),linux26)
-  ifneq (,$(wildcard $(ROOT_PREFIX)/etc/udev/rules.d))
+  ifneq (,$(wildcard $(INSTALL_PREFIX)/etc/udev/rules.d))
     DYNFS=yes
     UDEVRULES=yes
   endif
@@ -104,10 +111,10 @@ LIBTONEZONE_SO:=libtonezone.so
 LIBTONEZONE_SO_MAJOR_VER:=1
 LIBTONEZONE_SO_MINOR_VER:=0
 
-MODULES:=zaptel tor2 torisa wcusb wcfxo wctdm wctdm24xxp \
+MODULES:=zaptel tor2 torisa wcfxo wctdm wctdm24xxp \
 	 ztdynamic ztd-eth wct1xxp wcte11xp pciradio \
          ztd-loc # ztdummy
-#MODULES+=wcfxsusb
+#MODULES+=wcusb wcfxsusb
 # build ztdummy by default for 2.6 kernels
 ifeq ($(BUILDVER),linux26)
 MODULES+=ztdummy
@@ -126,14 +133,14 @@ obj-m+=wct4xxp/
 # This line is only meaningful when this Makefile is used as kconfig for 
 # 2.6 build
 
-ifneq (,$(shell [ 0$(SUBLEVEL) -ge 10 ] && [ "$(ARCH)" = 'i386' ] && echo 1))
-obj-m+=xpp/
-endif
+#ifneq (,$(shell [ 0$(SUBLEVEL) -ge 10 ] && [ "$(ARCH)" = 'i386' ] && echo 1))
+#obj-m+=xpp/
+#endif
 
 ifneq (,$(wildcard /usr/include/newt.h))
 ZTTOOL:=zttool
 endif
-BINS=ztcfg torisatool makefw ztmonitor ztspeed $(ZTTOOL) zttest fxotune
+BINS=ztcfg torisatool ztmonitor ztspeed $(ZTTOOL) zttest fxotune wct4xxp/fw2h
 
 all: $(BUILDVER) $(LIBTONEZONE_SO)
 
@@ -177,21 +184,21 @@ pciradio.o: radfw.h
 ztdummy.o: ztdummy.h
 
 $(MODULESO): %.o: %.c zaptel.h
-	$(CC) $(KFLAGS) -o $@ -c $<
+	$(HOSTCC) $(KFLAGS) -o $@ -c $<
 
 tor2ee.o: tor2-hw.h
 
 tor2ee: tor2ee.o
-	$(CC) $(CFLAGS) -o $@ $^ -lpci
+	$(HOSTCC) $(CFLAGS) -o $@ $^ -lpci
 
 zonedata.lo: zonedata.c
-	$(CC) -c $(LCFLAGS) -o $@ $^
+	$(HOSTCC) -c $(LCFLAGS) -o $@ $^
 
 tonezone.lo: tonezone.c
-	$(CC) -c $(LCFLAGS) -o $@ $^
+	$(HOSTCC) -c $(LCFLAGS) -o $@ $^
 
 torisatool: torisatool.o
-	$(CC) -o $@ $^
+	$(HOSTCC) -o $@ $^
 
 tones.h: gendigits
 	./gendigits > $@
@@ -202,8 +209,11 @@ tor2fw.h: makefw tormenta2.rbt
 radfw.h: makefw pciradio.rbt
 	./makefw pciradio.rbt radfw > radfw.h
 
-gendigits: gendigits.o
-	$(CC) -o $@ $^ -lm
+gendigits: gendigits.c
+	$(BUILDCC) -o $@ $^ -lm
+
+makefw: makefw.c
+	$(BUILDCC) -o $@ $^ -lm
 
 zaptel.c: tones.h 
 
@@ -216,66 +226,69 @@ ztprovision.o: ztprovision.c zaptel.h
 ztmonitor.o: ztmonitor.c zaptel.h
 
 ztspeed.o: ztspeed.c
-	$(CC) -o $@ -c $^
+	$(HOSTCC) -o $@ -c $^
 
 zttool: zttool.o
-	$(CC) -o $@ $^ -lnewt
+	$(HOSTCC) -o $@ $^ -lnewt
 
 ztmonitor: ztmonitor.o
-	$(CC) -o $@ $^
+	$(HOSTCC) -o $@ $^
 
 ztspeed: ztspeed.o
-	$(CC) -o $@ $^
+	$(HOSTCC) -o $@ $^
 
 sethdlc-new: sethdlc-new.o
-	$(CC) -o $@ $^
+	$(HOSTCC) -o $@ $^
 
 sethdlc-new.o: sethdlc-new.c
-	$(CC) -o $@ -c $(CFLAGS) -I$(KINCLUDES) $^
+	$(HOSTCC) -o $@ -c $(CFLAGS) -I$(KINCLUDES) $^
 
 libtonezone.a: $(TZOBJS)
-	ar rcs libtonezone.a $^
+	$(HOSTAR) rcs libtonezone.a $^
 
 $(LIBTONEZONE_SO): $(TZOBJS)
-	$(CC) -shared -Wl,-soname,$(LIBTONEZONE_SO).$(LIBTONEZONE_SO_MAJOR_VER).$(LIBTONEZONE_SO_MINOR_VER) -lm -o $@ $^
+	$(HOSTCC) -shared -Wl,-soname,$(LIBTONEZONE_SO).$(LIBTONEZONE_SO_MAJOR_VER).$(LIBTONEZONE_SO_MINOR_VER) -lm -o $@ $^
 
 ztcfg.c: ztcfg.h
 
 ztcfg-shared: ztcfg.o $(LIBTONEZONE_SO)
-	$(CC) -o $@ $^ -lm
+	$(HOSTCC) -o $@ $^ -lm
 
 ztcfg: ztcfg.o libtonezone.a
-	$(CC) -o $@ $^ -lm
+	$(HOSTCC) -o $@ $^ -lm
 
 ztcfg-dude: ztcfg-dude.o mknotch.o complex.o $(LIBTONEZONE_SO)
-	$(CC) -o $@ $^ -lm
+	$(HOSTCC) -o $@ $^ -lm
 
 mknotch.o: mknotch.cc
-	$(CC) -o $@ -c $^
+	$(HOSTCC) -o $@ -c $^
 
 complex.o: complex.cc
-	$(CC) -o $@ -c $^
+	$(HOSTCC) -o $@ -c $^
 
 usbfxstest.o: usbfxstest.c
-	$(CC) -o $@ -g -c $^
+	$(HOSTCC) -o $@ -g -c $^
 
 usbfxstest: usbfxstest.o 
-	$(CC) -o $@ $^ -lzap
+	$(HOSTCC) -o $@ $^ -lzap
 
 fxstest: fxstest.o $(LIBTONEZONE_SO)
-	$(CC) -o $@ $^ -lm
+	$(HOSTCC) -o $@ $^ -lm
 
 fxotune: fxotune.o
-	$(CC) -o $@ $^ -lm
+	$(HOSTCC) -o $@ $^ -lm
 
 fxsdump: fxsdump.o
-	$(CC) -o $@ $^ -lm
+	$(HOSTCC) -o $@ $^ -lm
 
 stackcheck: checkstack $(BUILDVER)
 	./checkstack *.o
 
 ztdiag: ztdiag.o 
-	$(CC) -o $@ $^
+	$(HOSTCC) -o $@ $^
+
+wct4xxp/fw2h: wct4xxp/fw2h.c
+	$(BUILDCC) -o $@ $^ -lm
 
 devices:
 ifndef DYNFS
@@ -348,14 +361,14 @@ endif
 	install -D -m 644 zaptel.h $(INSTALL_PREFIX)/usr/include/linux/zaptel.h
 	install -D -m 644 torisa.h $(INSTALL_PREFIX)/usr/include/linux/torisa.h
 	install -D -m 644 tonezone.h $(INSTALL_PREFIX)/usr/include/tonezone.h
-	install -m 644 doc/ztcfg.8 $(INSTALL_PREFIX)/usr/share/man/man8
-	install -m 644 doc/zttool.8 $(INSTALL_PREFIX)/usr/share/man/man8
+	#install -m 644 doc/ztcfg.8 $(INSTALL_PREFIX)/usr/share/man/man8
+	#install -m 644 doc/zttool.8 $(INSTALL_PREFIX)/usr/share/man/man8
 	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
-	[ -f $(CONFIG_FILE) ] || install -D -m 644 zaptel.conf.sample $(CONFIG_FILE)
-	build_tools/genmodconf $(BUILDVER) "$(ROOT_PREFIX)" "$(filter-out zaptel,$(MODULES))"
-	@if [ -d /etc/modutils ]; then \
-		/sbin/update-modules ; \
-	fi
+	[ -f $(INSTALL_PREFIX)$(CONFIG_FILE) ] || install -D -m 644 zaptel.conf.sample $(INSTALL_PREFIX)$(CONFIG_FILE)
+	#build_tools/genmodconf $(BUILDVER) "$(ROOT_PREFIX)" "$(filter-out zaptel,$(MODULES))"
+	#@if [ -d /etc/modutils ]; then \
+	#	/sbin/update-modules ; \
+	#fi
 
 install-udev: devices
 
diff -Nurp zaptel-1.2.10.orig/wct4xxp/Kbuild zaptel-1.2.10/wct4xxp/Kbuild
--- zaptel-1.2.10.orig/wct4xxp/Kbuild	2006-08-27 04:02:42.000000000 +0200
+++ zaptel-1.2.10/wct4xxp/Kbuild	2006-11-06 17:44:49.000000000 +0100
@@ -1,8 +1,5 @@
 # yes, this is redundant... the Kbuild system is changing to hostprogs-y,
 # but we need to be able to support older verions as well
-host-progs := fw2h
-hostprogs-y := fw2h
-
 obj-m += wct4xxp.o
 
 EXTRA_CFLAGS := -I$(src)/.. $(shell $(src)/../oct612x/octasic-helper cflags $(src)/../oct612x) -Wno-undef
diff -Nurp zaptel-1.2.10.orig/wct4xxp/Makefile zaptel-1.2.10/wct4xxp/Makefile
--- zaptel-1.2.10.orig/wct4xxp/Makefile	2006-08-27 04:27:23.000000000 +0200
+++ zaptel-1.2.10/wct4xxp/Makefile	2006-11-06 17:31:56.000000000 +0100
@@ -10,7 +10,7 @@ OCTASIC_CFLAGS:=$(shell ../oct612x/octas
 all: wct4xxp.o
 
 %.o: %.c
-	$(CC) $(KFLAGS) $(OCTASIC_CFLAGS) -o $@ -c $<
+	$(HOSTCC) $(KFLAGS) $(OCTASIC_CFLAGS) -o $@ -c $<
 
 base.o: ../zaptel.h vpm450m.h wct4xxp.h
 
@@ -19,8 +19,6 @@ vpm450m.o: vpm450m.h vpm450m_fw.h ../oct
 wct4xxp.o: base.o vpm450m.o $(OCTASIC_OBJS)
 	$(LD) -r -o $@ $^
 
-fw2h: CFLAGS=
-
 vpm450m_fw.h: OCT6114-128D.ima fw2h
 	./fw2h $< $@
 
