diff -ruN -x Makefile.in zlib-1.1.4.orig/configure zlib-1.1.4/configure
--- zlib-1.2.1.orig/configure   Sun Jul  9 16:26:39 2000
+++ zlib-1.2.1/configure        Sun Jul  9 16:26:26 2000
@@ -69,7 +69,7 @@
 
 if test "$gcc" -eq 1 && ($cc -c $cflags $test.c) 2>/dev/null; then
   CC="$cc"
-  SFLAGS=${CFLAGS-"-fPIC -O3"}
+  SFLAGS=${CFLAGS-"-D_REENTRANT -fPIC -O3"}
   CFLAGS="$cflags"
   case `(uname -s || echo unknown) 2>/dev/null` in
   Linux | linux) LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1"};;
